analytics_dashboard/core/middleware.py

```diff
@@ -8,10 +8,24 @@
 from django.template.response import TemplateResponse
 
 from core.exceptions import ServiceUnavailableError
+from waffle import flag_is_active
 
 logger = logging.getLogger(__name__)
 
 
+# For https://tasks.opencraft.com/browse/SE-3950 . If an upstream solution is invented by the time you're rebasing this,
+# remove it and the line that includes it from the middleware settings.
+class LanguageHeaderPreemptMiddleware(object):
+    """
+    If the appropriate waffle flag is present, ignore the 'Accept-Language' header.
+    """
+    def process_request(self, request):
+        if not flag_is_active(request, 'ignore_accept_language'):
```

Comment:
I would prefer if you use a better name for the waffle switch/flag, especially since we're going to be upstreaming this middleware. Something of the sorts of `analytics_dashboard.use_default_language` or `analytics_dashboard.ignore_browser_language`.

Also, when defining the Waffle Switch/Flag in the `config.py` or `config/waffle.py`, please use [this template](https://edx.readthedocs.io/projects/edx-toggles/en/latest/how_to/documenting_new_feature_toggles.html#boilerplate-template) to document it.

Reply:
Thanks for showing me that boilerplate. I was searching around looking for something like that and couldn't find anything.